### PR TITLE
chore: migrate from NPM to Yarn 3

### PR DESCRIPTION
**Pull request by antouhou**
## Issue being fixed or feature implemented

NPM has many downsides comparing with Yarn
- NPM workspaces require to specify current versions in dependencies which makes release process very tricky
- NPM install is very slow
- Yarn 3 provides many convenient features

## What was done?

- Switched from NPM to Yarn 3
- Simplified docker build logic
- Use Yarn constraints instead of syncpack to validate workspaces
- Use [corepack](https://nodejs.org/api/corepack.html)
- Defined resolutions to stick on specific deps

## How Has This Been Tested?

Manually and with CI

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: .github/workflows/dapi-grpc.yml [hunk 1]
```diff
@@ -15,6 +15,6 @@ on:
 jobs:
   dapi-tests:
     name: Run DAPI gRPC tests
-    uses: dashevo/platform/.github/workflows/test.yml@master
+    uses: dashevo/platform/.github/workflows/test.yml@feat-yarn2
```
Comment (strophy) [inline .github/workflows/dapi-grpc.yml:18]: I think this needs to change to `master` before merge?

Comment (shumkov) [reply .github/workflows/dapi-grpc.yml:18]: Yep!

Comment (antouhou) [reply .github/workflows/dapi-grpc.yml:18]: I think you can't rename it _before_ merge, as `master` will have the correct file only after the merge? @shumkov

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -49,7 +48,7 @@ jobs:
             return (tag.includes('dev') ? `${major}.${minor}-dev` : 'latest');
 
       - name: Publish NPM packages
-        run: npm publish --workspaces --access public --tag ${{ steps.tag.outputs.result }}
+        run: yarn npm publish --access public --tag ${{ steps.tag.outputs.result }}
```
Comment (antouhou) [inline .github/workflows/release.yml:51]: Will this actually work?

File: README.md [hunk 3]
```diff
@@ -38,13 +38,14 @@ located in the [packages](./packages) directory.
...
   - [docker-compose](https://docs.docker.com/compose/install/) v1.29.2+
-- Run `npm run setup` to install dependencies and configure and build all packages
-- Run `npm run start` to start the local dev environment built from the sources
-- Run `npm test` to run the whole test suite (note that running tests requires a running node, 
- so be sure to call `npm run start` first). Alternatively, you can run tests for a specific 
+- Run `corepack enable` to enable [corepack](https://nodejs.org/dist/latest/docs/api/corepack.html) and install yarn
+- Run `yarn setup` to install dependencies and configure and build all packages
+- Run `yarn start` to start the local dev environment built from the sources
+- Run `yarn test` to run the whole test suite (note that running tests requires a running node, 
+ so be sure to call `npm start` first). Alternatively, you can run tests for a specific
```
Comment (antouhou) [inline README.md:48]: `yarn start`

File: packages/dapi/Dockerfile [hunk 4]
```diff
@@ -1,37 +1,25 @@
 # syntax = docker/dockerfile:1.3
-FROM node:16-alpine as node_gyp_cache
-
-RUN apk update && \
-    apk --no-cache upgrade && \
-    apk add --no-cache git \
-                      openssh-client \
-                      python3 \
-                      alpine-sdk \
-                      zeromq-dev
```
Comment (antouhou) [inline packages/dapi/Dockerfile:10]: We don't need to install the zmq dev package anymore?

Comment (antouhou) [reply packages/dapi/Dockerfile:10]: Ah, I see, it was just moved to a different place

File: .github/workflows/dapi-grpc.yml (hunk 1 above)
Comment (shumkov) [reply .github/workflows/dapi-grpc.yml:18]: Yeah, but before we need to make sure that tests are running, which is not a case atm.

File: .github/workflows/release.yml (hunk 2 above)
Comment (shumkov) [reply .github/workflows/release.yml:51]: Honestly, I didn't try it yet :)

Comment (shumkov) [reply .github/workflows/release.yml]: Now it should work.

File: README.md (hunk 3 above)
Comment (shumkov) [reply README.md]: Thank you. Changed all npm mentions.

**Pull request merged**